tests/integration/test_lepy.py

```diff
@@ -0,0 +1,60 @@
+import os
+import subprocess
+import time
+
+import pytest
+import requests
+from lepy import run_lego_command
+
+
+@pytest.fixture(scope="session")
+def configure_acme_server():
+    """Get and install pebble, a lightweight ACME server from letsencrypt."""
+    tests_dir = os.path.dirname(__file__)
+
+    # Install and run pebble
+    subprocess.check_call(["go", "install", "./cmd/pebble"], cwd=os.path.join(tests_dir, "pebble"))
+    pebble = subprocess.Popen(
+        ["pebble", "-config", "test/config/pebble-config.json"],
+        cwd=os.path.join(tests_dir, "pebble"),
+    )
+
+    ca_path = os.path.join(tests_dir, "pebble/test/certs/pebble.minica.pem")
+    filename = os.path.join(tests_dir, "test_files/test.csr")
+    localhost_csr = open(filename).read().encode()
+
+    poll_server("https://0.0.0.0:14000/dir")
+
+    yield {"csr": localhost_csr, "ca_path": ca_path}
+
+    pebble.terminate()
+
+
+class TestLepy:
+    def test_given_request_certificate_when_request_sent_then_certificate_issued(
+        self,
+        configure_acme_server,
+    ):
+        response = run_lego_command(
+            "something@nowhere.com",
+            "https://localhost:14000/dir",
+            configure_acme_server.get("csr"),
+            "http",
+            {
+                "SSL_CERT_FILE": configure_acme_server.get("ca_path"),
+                "HTTP01_PORT": "5002",
+                "TLSALPN01_PORT": "5001",
+            },
+        )
+        assert response.metadata.domain == "localhost"
+
+
+def poll_server(url: str, freq: int = 1):
+    while True:
+        try:
+            time.sleep(freq)
+            response = requests.get(url, verify=False)
+            return response
+        except requests.RequestException as e:
+            print(e)
+            pass
```

Comment:
If for some reason pebble fails, won't this continue forever? Maybe we should add a reasonable timeout here?

Reply:
if pebble fails, the `.Popen` call will fail before we can get to this point. This would only go forever if pebble succeeded but for some reason not responding. I still added a timeout.